mapie/quantile_regression.py

```diff
@@ -0,0 +1,323 @@
+                sample_weight_fit, sample_weight_calib) = train_test_split(
+                X,
+                y,
+                sample_weight,
+                test_size=0.5,
+                random_state=42
+            )
+        list_estimators = []
+        list_y_preds_calib = []
+        name_estimator = estimator.__class__.__name__
+        for item in alpha_copy:
+            estimator_cloned = clone(estimator)
+            params = {
+                self.link_estimator_quantile[name_estimator][1]: item
+            }
+            estimator_cloned.set_params(**params)
+            estimator_cloned_ = fit_estimator(
+                estimator_cloned, X_fit, y_fit, sample_weight_fit
+            )
+            list_estimators.append(estimator_cloned_)
+            y_pred_calib = estimator_cloned_.predict(X_calib)
+            # check_nan_in_aposteriori_prediction(y_pred_calib)
+            list_y_preds_calib.append(y_pred_calib)
+            self.y_calib = y_calib
+
+        self.list_y_preds_calib = list_y_preds_calib
+        self.list_estimators = list_estimators
+
+        self.conformity_scores_ = np.full(
+                shape=(3, len(y_calib)),
+                fill_value=np.nan
+            )
+        self.conformity_scores_[0] = self.list_y_preds_calib[0]-self.y_calib
+        self.conformity_scores_[1] = self.y_calib-self.list_y_preds_calib[1]
+        self.conformity_scores_[2] = np.max(
+                [self.conformity_scores_[0],
+                 self.conformity_scores_[1]],
+                axis=0)
+        return self
+
+    def predict(
+        self,
+        X: ArrayLike,
+        ensemble: bool = False,
+        symmetry: Optional[bool] = True,
+    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:
+
+        # Checks
+        check_is_fitted(self, self.fit_attributes)
+        self._check_ensemble(ensemble)
+        alpha = cast([NDArray], check_alpha(self.alpha))
+        alpha_np = cast(NDArray, alpha)
+        n = len(self.y_calib)
+        check_alpha_and_n_samples(alpha_np, n)
+
+        y_preds = np.full(
+                shape=(3, X.shape[0]),
+                fill_value=np.nan
+            )
+        n = len(self.y_calib)
+        q = (1-(alpha_np/2))*(1+(1/n))
+
+        y_preds[0] = self.list_estimators[0].predict(X)
+        y_preds[1] = self.list_estimators[1].predict(X)
+        y_preds[2] = self.list_estimators[2].predict(X)
+        if symmetry is True:
+            q = (1-(alpha_np))*(1+(1/n))
+            quantile = np.full(2, np_quantile(
+                self.conformity_scores_[2], q, method="higher"
+            ))
+        else:
+            quantile = np.full(shape=2, fill_value=np.nan)
+            quantile[0] = np_quantile(
+                self.conformity_scores_[0], q, method="higher"
+            )
+            quantile[1] = np_quantile(
+                self.conformity_scores_[1], q, method="higher"
+            )
+        y_pred_low = y_preds[0][:, np.newaxis] - quantile[0]
```

Comment:
Why do we need to to create a new axis here ?

Reply:
OK agreed. In practice, the last dimension will be always equal to 1 since one `alpha` is provided at a time.